modules/experiment/src/main/java/com/intuit/wasabi/experiment/impl/PagesImpl.java

```diff
@@ -206,4 +215,13 @@ public void erasePageData(Application.Name applicationName, Experiment.ID experi
         }
         return pagesRepository.getPageExperimentList(applicationName);
     }
+
+    @Override
+    public List<PageExperiment> getExperimentsWithoutLabels(Application.Name applicationName, Page.Name pageName) {
+        // Throw an exception if application name is invalid
+        if (applicationName == null || StringUtils.isBlank(applicationName.toString())){
```

Comment:
Other methods in the PagesImpl seems to be expecting a NULL applicationName so keeping the NULL check with the isNull()...